chainio/clients/avsregistry/reader.go

```diff
@@ -119,25 +119,30 @@ func BuildAvsRegistryChainReader(
 ) (*AvsRegistryChainReader, error) {
 	contractRegistryCoordinator, err := regcoord.NewContractRegistryCoordinator(registryCoordinatorAddr, ethClient)
 	if err != nil {
+		logger.Error("Failed to create contractRegistryCoordinator", "err", err)
```

Comment:
should we rather wrap the error with informative text and then return it and not log here? reason it's client responsibility to log errors otherwise we have too many logs?

Reply:
Yeah I was thinking so too. But we've been doing it this way everywhere else so wanted to stay consistent. Shall we change it everywhere?

Reply:
yea we should maybe do this ? https://pkg.go.dev/errors#Join 

for ex: `errors.Join(errors.new("Failed to create contractRegistryCoordinator", err))`

Reply:
I agree with @shrimalmadhur , you should return the wrapped error and the client should handle it.

Reply:
> yea we should maybe do this ? https://pkg.go.dev/errors#Join
> 
> for ex: `errors.Join(errors.new("Failed to create contractRegistryCoordinator", err))`

join literally concatenates the error strings with a newline in between, such that printing them prints the 2 errors on two separate lines. Is that fine with you @uri-lightblocks ? I know you didn't like the newlines in avs-sync.

Reply:
I think new line is okay in the error object? not sure how will it behave in elk system when someone logs `err.Errror()` - I wonder if we should have custom error wrapping but I am scope creeping here 😅

Reply:
ok changed here: [856915e](https://github.com/Layr-Labs/eigensdk-go/pull/115/commits/856915ec9b42b9ecb4fd9666a73f9ad74c2a13fe)

Reply:
@uri-lightblocks tell me more man. How would you concatenate it? With a space in between? A tilda? A "-"? :)

Reply:
Let's get this merged. At least now we can easily change it whenever.